### ISSUE TITLE
missing file extension error 

### ISSUE THREAD

**Issue by rbalicki2** [benmosher/eslint-plugin-import#991]
Love the package! I believe this isn't addressed by any of the other issues, although I only did a quick search for query.

When I import `import maskIcon from 'src/components/Root/favicons/safari-pinned-tab.svg?external';` (I'm using a resource query because most svg's go through react-svg-loader, but this one goes through file-loader), I get the following lint error:

```
  19:22  error  Missing file extension "svg" for "src/components/Root/favicons/safari-pinned-tab.svg?external"  import/extensions
```

Thank you! And happy new year 🎉

**Comment by ljharb** [comment]
In this case, you're not importing a normal file (especially since svgs can't be imported, normally) - you'd need to configure the plugin with a webpack-specific resolver so it knows how to parse these paths.

**Comment by rbalicki2** [comment]
It works without the resource query. What do you mean by configure a webpack specific resolver? I have eslint-import-resolver-webpack, and loaders set up for svg files.

**Comment by ljharb** [comment]
hm - if that's the case, then probably the resolver needs to be able to provide the "file without extension" instead of the logic [here](https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/extensions.js#L114).

**Comment by IanEdington** [comment]
This seems to be related: https://github.com/tleunen/eslint-import-resolver-babel-module/pull/60

Also, I'm having this issue with other file types. All using query parameters for webpack.

**Comment by dgautsch** [comment]
I took some time to review this issue and from what i can tell `eslint-plugin-import` is handling query strings correctly and I can't recreate the issue in the unit tests (albeit it's a contrived scenario). My theory is that something is processing the node and stripping the extension/querystring before it reaches `eslint-plugin-import`. Will keep investigating.

**Comment by ljharb** [comment]
This has since been fixed; in v2.20.0. Happy to reopen if not.
